Feat apply ruleslist

Hugo-Barroca opened the pull request:
Unit tests will be added in another Pull Request.

Hugo-Barroca committed:
Created a column in the check quality GUI for each custom rule (WIP).

Hugo-Barroca committed:
Added javadocs, cleaned code.

Hugo-Barroca committed:
Added javadocs, cleaned code.

Hugo-Barroca committed:
Added a catch verification.

franfaccin commented on src/main/java/controller/MainController.java:
```diff
@@ -1,241 +1,324 @@
+		colNames[1] = "PMD";
+		colNames[2] = "iPlasma";
+		colNames[3] = "long_method";
+		colNames[4] = "feature_envy";
```
Já que comparamos tudo com esses 2 valores, o que achas de eles estarem antes do PMD e iPlasma? Então a ordem seria: Method ID, long, envy, PMD, iPlasma, outros.

Hugo-Barroca replied:
Parece-me bem, troquei a ordem agora 👍

franfaccin commented on src/main/java/controller/MainController.java:
```diff
@@ -1,241 +1,324 @@
+	 * @return An Array of String arrays where each line is a row with the code
+	 *         quality results for a method, and each column is the value of that
+	 *         result line for that column
+	 */
```
Além de descrever o retorno, descreve tbm o método logo antes, algo tipo

```java
/** Esse método faz isso
 * @return String[][] An Array of String arrays ...
 */
```

franfaccin commented on src/main/java/controller/MainController.java:
```diff
@@ -1,241 +1,324 @@
+		float LAA = row.getLAA();
+		String filledRule = "var ATFD = " + ATFD + "; " + "var CYCLO = " + CYCLO + "; " + "var LOC = " + LOC + "; "
+				+ "var LAA = " + LAA + "; ";
+		engine.eval(filledRule);
```
Como falamos, veja como fica se retornares a `filledRule` nesse método. E usa a engine só uma vez dentro do `getResult`, quem sabe ajuda na velocidade de rodar o código, um uso a menos da engine por linha por regra.

Hugo-Barroca committed:
Changed column order in the check quality popup.

Hugo-Barroca committed:
Fixed the results' order in the check quality popup.

Hugo-Barroca committed:
Changed the registerVariables() method.

The registerVariables() method will now return a String, with all of the
necessary javascript statements for metric variable initialization. This
string should be passed on to the JavaScript engine as necessary.

Hugo-Barroca committed:
Cleaned javaDocs and the registerVariables() method.

franfaccin commented on src/main/java/controller/MainController.java:
```diff
@@ -1,241 +1,325 @@
+				checkCodeQualityAndShow();
+			} catch (Exception e2) {
+				qualityGui.hide();
+				JOptionPane.showMessageDialog(null, "Invalid rule syntax! Please verify!");
```
Por favor, adiciona o nome da regra que falhou!

Hugo-Barroca committed:
Changed the invalid rule warning, deleted stray try/catch.

Hugo-Barroca committed:
Eddited invalid rule warning, for clarity.

vandabarata commented on src/main/java/controller/MainController.java:
```diff
@@ -1,241 +1,331 @@
+		colNames[1] = "long_method";
+		colNames[2] = "feature_envy";
+		colNames[3] = "PMD";
+		colNames[4] = "iPlasma";
```
why are we hardcoding this? can't we get them from the real excel?

Hugo-Barroca replied:
We could, but we'd have to hardcode the sheet, row and column number of the cell where we'd get this information from. If that's okay, I'll change it around.

vandabarata commented on src/main/java/controller/MainController.java:
```diff
@@ -1,241 +1,331 @@
+		for (ExcelRow row : excelRowsConverted) {
+			String[] qualityRow = new String[4 + excelRowsConverted.size()];
+			qualityRow[0] = Integer.toString(row.getId());
+			qualityRow[1] = Boolean.toString(row.isIs_long_method());
```
this is missing the refactor right? the one diego did?

Hugo-Barroca replied:
It's possible, this branch was created a couple of days ago. Which variables did he refactor?

vandabarata replied:
basically everything - it's already in the master branch, you can merge into yours and solve the conflicts

vandabarata commented on src/main/java/controller/MainController.java:
```diff
@@ -1,241 +1,331 @@
+					qualityGui.hide();
+					wrongRule = rule;
+					JOptionPane.showMessageDialog(null,
+							"Invalid rule syntax! Please verify the rule with the name \"" + wrongRule + "\"!");
```
```suggestion
							"Invalid rule syntax! Please verify the conditions for the rule  \"" + wrongRule + "\"!");
```

Hugo-Barroca committed:
Fixed code redundancy.

vandabarata commented on src/main/java/controller/MainController.java:
```diff
@@ -1,241 +1,331 @@
+		int LOC = row.getLOC();
+		float LAA = row.getLAA();
+		String filledRule = "var ATFD = " + ATFD + "; " + "var CYCLO = " + CYCLO + "; " + "var LOC = " + LOC + "; "
+				+ "var LAA = " + LAA + "; ";
```
@ftfni-iscteiul tem que ser var wtv; var wtv2; ...? não dá para fazer continuous no js? tipo var wtv, wtv2, wtv3...? se desse, acho que ficaria melhor assim

vandabarata commented on src/main/java/controller/MainController.java:
```diff
@@ -1,241 +1,331 @@
+	 * which initializes all the necessary metric variables.
+	 * 
+	 * @param row The excel row with the values for our metrics.
+	 * @return filledRule The Sting of rule initializations to be passed on to a
```
the Sting! oh noes! also, it doesn't return rule initializations. it returns the metrics turned into variables to use in the JS engine for running the rules

Hugo-Barroca committed:
Changed the hardcoded values for the column names to their excel values.

Hugo-Barroca committed:
Small change to the invalid rule warning.

Hugo-Barroca committed:
Fixed JDocs.

vandabarata commented on src/main/java/controller/MainController.java:
```diff
@@ -206,11 +206,11 @@ private void checkCodeQualityAndShow() {
+		colNames[1] = excelRows.get(0)[8];
+		colNames[2] = excelRows.get(0)[11];
+		colNames[3] = excelRows.get(0)[10];
+		colNames[4] = excelRows.get(0)[9];
```
:+1:

Hugo-Barroca committed:
Merge remote-tracking branch 'origin/master' into feat-apply-ruleslist

Conflicts:
	src/main/java/controller/MainController.java

Hugo-Barroca committed:
Changed the way the JS variables are initialized.

Hugo-Barroca commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(3) [Story] Aplicar a rulesList em cada methodID e apresentar na GUI "Quality Code Results" os resultados](https://trello.com/c/HFrhJxvJ/3-3-story-aplicar-a-ruleslist-em-cada-methodid-e-apresentar-na-gui-quality-code-results-os-resultados)

Hugo-Barroca commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(3) [Story] Aplicar a rulesList em cada methodID e apresentar na GUI "Quality Code Results" os resultados](https://trello.com/c/HFrhJxvJ/3-3-story-aplicar-a-ruleslist-em-cada-methodid-e-apresentar-na-gui-quality-code-results-os-resultados)

franfaccin commented on src/main/java/controller/MainController.java:
```diff
@@ -202,25 +210,103 @@ private void addButton(JButton addButton) {
+		colNames[1] = excelRows.get(0)[8];
+		colNames[2] = excelRows.get(0)[11];
+		colNames[3] = excelRows.get(0)[10];
+		colNames[4] = excelRows.get(0)[9];
```
Eu preferia antes na verdade hahah Magic numbers é pior que o texto hardcoded. Se quer usar número, eu preferia umas variáveis tipo
```
final int METHOD_ID_INDEX = 0;
final int PMD_INDEX = 8;
final int IPLASMA_INDEX = 11;
//...
colNames[0] = excelRows.get(0)[METHOD_ID_INDEX];
colNames[1] = excelRows.get(0)[PMD_INDEX];
colNames[2] = excelRows.get(0)[IPLASMA_INDEX];
//...
```
E se vc ver que dei nomes errados pros índices, é justamente pq quem não fez não faz ideia do que são esses valores xD

vandabarata replied:
eu ia dizer que ele podia comentar a frente o que sao... mas isso também dá, sim x) até acho que é a melhor solução

Hugo-Barroca committed:
Added constants for the rule names in the header columns.

jmalo1-iscteiul reviewed (approved):
Seems all good to me!

vsfba1-iscteiul commented:
Approved pending resolution of merge conflicts

Hugo-Barroca committed:
Merge branch 'master' into feat-apply-ruleslist

Hugo-Barroca commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(1) [Task] Lançar uma exceção caso haja falhas no Runtime da regra e abrir um popup a indicar a regra em falha e indicar que deve corrigi-la [1]](https://trello.com/c/dyyiyi5V/2-1-task-lan%C3%A7ar-uma-exce%C3%A7%C3%A3o-caso-haja-falhas-no-runtime-da-regra-e-abrir-um-popup-a-indicar-a-regra-em-falha-e-indicar-que-deve-co)

Hugo-Barroca commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(1) [Task] Lançar uma exceção caso haja falhas no Runtime da regra e abrir um popup a indicar a regra em falha e indicar que deve corrigi-la [1]](https://trello.com/c/dyyiyi5V/2-1-task-lan%C3%A7ar-uma-exce%C3%A7%C3%A3o-caso-haja-falhas-no-runtime-da-regra-e-abrir-um-popup-a-indicar-a-regra-em-falha-e-indicar-que-deve-co)
